File: app/src/main/java/igrek/songbook/service/preferences/PropertyType.java
package igrek.songbook.service.preferences;

enum PropertyType {
	
	STRING(String.class),
	
	BOOLEAN(Boolean.class),
	
	INTEGER(Integer.class),
	
	LONG(Long.class),
	
	FLOAT(Float.class);
	
	private Class<?> clazz;
	
	PropertyType(Class<?> clazz) {
		this.clazz = clazz;
	}
	
	public Class<?> getClazz() {
		return clazz;
	}
}

File: app/src/main/java/igrek/songbook/settings/preferences/PropertyType.kt
package igrek.songbook.settings.preferences

enum class PropertyType constructor(val clazz: Class<*>) {

    STRING(String::class.javaObjectType),

    BOOLEAN(Boolean::class.javaObjectType),

    INTEGER(Int::class.javaObjectType),

    LONG(Long::class.javaObjectType),

    FLOAT(Float::class.javaObjectType)
}
